www/Dockerfile

```diff
@@ -1,3 +1,14 @@
+FROM alpine AS oqs-provider
+
+RUN apk add --no-cache cmake git g++ make ninja openssl openssl-dev
+
+RUN git clone --depth 1 --branch 0.12.0 https://github.com/open-quantum-safe/liboqs.git
+RUN git clone --depth 1 --branch 0.8.0 https://github.com/open-quantum-safe/oqs-provider.git
```

Comment:
To avoid hard coding these, I would use a [build arg](https://docs.docker.com/build/building/variables/). Then you can do something like:

```suggestion
ARG LIBOQS_VERSION=""
ARG OQSPROVIDER_VERSION=""

RUN if [ -z ${LIBOQS_VERSION} ]; exit 1; fi;
RUN if [ -z ${OQSPROVIDER_VERSION} ]; exit 2; fi;

RUN git clone --depth 1 --branch ${LIBOQS_VERSION} https://github.com/open-quantum-safe/liboqs.git
RUN git clone --depth 1 --branch ${OQSPROVIDER_VERSION} https://github.com/open-quantum-safe/oqs-provider.git
```

Then as part of the build step, or if you are using docker compose you can define it in the yaml file, you can define the version you want. The above should cause a build error if the versions aren't defined.